core/src/calleeFunctions/index.ts

```diff
@@ -23,27 +23,101 @@ const allCalleeFunctions: Record<CalleeNames, CalleeFunctions> = {
 export const getCalleeData = async function (
     network: string,
     collateralType: string,
+    marketId: string,
     profitAddress: string
 ): Promise<string> {
     const collateral = getCollateralConfigByType(collateralType);
-    const calleeFuctions = allCalleeFunctions[collateral.exchange.callee];
-    if (!calleeFuctions) {
+    const marketData = collateral.exchanges[marketId];
+    if (!marketData || !marketData.callee || !allCalleeFunctions[marketData.callee]) {
         throw new Error(`Unsupported collateral type "${collateralType}"`);
     }
-    return await calleeFuctions.getCalleeData(network, collateral, profitAddress);
+    return await allCalleeFunctions[marketData.callee].getCalleeData(network, collateral, marketId, profitAddress);
 };
 
-const _getMarketPrice = async function (
+export const getMarketData = async function (
     network: string,
     collateralSymbol: string,
     amount: BigNumber = new BigNumber('1')
-): Promise<BigNumber> {
+): Promise<Record<string, MarketData>> {
     const collateral = getCollateralConfigBySymbol(collateralSymbol);
-    const calleeFuctions = allCalleeFunctions[collateral.exchange.callee];
-    if (!calleeFuctions) {
+    const isCollateralSupported = Object.values(collateral.exchanges).some(
+        marketData => marketData.callee && allCalleeFunctions[marketData.callee]
+    );
+    if (!isCollateralSupported) {
         throw new Error(`Unsupported collateral symbol "${collateralSymbol}"`);
     }
-    return await calleeFuctions.getMarketPrice(network, collateral, amount);
+    let marketDataRecords = {};
+    for (const marketId in collateral.exchanges) {
+        const marketData = collateral.exchanges[marketId];
+        let marketUnitPrice: BigNumber;
+        try {
+            marketUnitPrice = await allCalleeFunctions[marketData.callee].getMarketPrice(
+                network,
+                collateral,
+                marketId,
+                amount
+            );
+        } catch {
+            marketUnitPrice = new BigNumber(NaN);
+        }
+        if (marketData.callee === 'UniswapV2LpTokenCalleeDai') {
+            marketDataRecords = {
+                ...marketDataRecords,
+                [marketId]: {
+                    marketUnitPrice,
+                    token0: marketData.token0,
+                    token1: marketData.token1,
+                },
```

Comment:
Don't you need to add other fields from `marketData` here as well?
```suggestion
                [marketId]: {
                    ...marketData,
                    marketUnitPrice,
                    token0: marketData.token0,
                    token1: marketData.token1,
                },
```